criu/pie/restorer.c

```diff
@@ -1521,6 +1521,31 @@ int cleanup_current_inotify_events(struct task_restore_args *task_args)
 	return 0;
 }
 
+/*
+ * Restore membarrier() registrations.
+ */
+static int restore_membarrier_registrations(int mask)
+{
+	unsigned long bitmap[1] = { mask };
+	int i, failed_mask = 0;
+
+	if (!mask)
+		return 0;
+
+	pr_info("Restoring membarrier() registrations %x\n", mask);
+
+	for_each_bit(i, bitmap)
+		if (sys_membarrier(1 << i, 0, 0))
+			failed_mask |= 1 << i;
```

Comment:
we need to print a return code in the error message, otherwise it will be impossible to determine a reason of an error.